pkg/sql/catalog/tabledesc/table_desc_builder.go

```diff
@@ -390,7 +390,6 @@ func maybeFillInDescriptor(
 	}
 	set(catalog.UpgradedPrivileges, fixedPrivileges)
 	set(catalog.RemovedDuplicateIDsInRefs, maybeRemoveDuplicateIDsInRefs(desc))
-	set(catalog.SetCheckConstraintColumnIDs, maybeSetCheckConstraintColumnIDs(desc))
```

Comment:
will this have the same issue with old text fixtures as when we removed `maybeAddConstraintIDs`? should we just move this to `RestoreChanges`

Reply:
i kicked off a test of tpchvec/direct_scans based on this PR to make sure it works still: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14341371?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true

let's see if it passes. i would prefer not to keep the code if there's no need.

Reply:
restoring from a 21.1 backup already has a unit test here, and it passed: https://github.com/cockroachdb/cockroach/blob/8cd23eb1d55d59d6028f3902e6f305e66acb6316/pkg/ccl/backupccl/testdata/backup-restore/show_backup#L7